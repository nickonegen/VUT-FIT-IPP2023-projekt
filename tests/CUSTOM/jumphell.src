.IPPcode23

# jumphell.src for testing STATP

# There are 13 jumps - 4 fwjumps, 5 backjumps and 4 badjumps;
# 50 total lines, 38 with instructions, 21 with comments

LABEL start						# label 1
PUSHS int@5
PUSHS int@10
DEFVAR GF@kek
DEFVAR GF@lol
DEFVAR GF@result
POPS GF@lol
POPS GF@kek
ADD GF@result GF@kek GF@lol
PUSHS GF@result
JUMPIFEQ label1 GF@result int@15		# jump 1, fwjump 1
JUMPIFNEQ badjump1 GF@result int@15	# jump 2, badjump 1

LABEL label1						# label 2
POPS GF@kek
MOVE GF@lol int@3
SUB GF@result GF@kek GF@lol
JUMPIFEQ label2 GF@result int@12		# jump 3, fwjump 2
JUMPIFNEQ badjump2 GF@result int@12	# jump 4, badjump 2

LABEL label3						# label 3
MOVE GF@kek int@2
MOVE GF@lol int@3
MUL GF@result GF@kek GF@lol
JUMPIFEQ end GF@result int@6			# jump 5, fwjump 3
JUMPIFNEQ badjump3 GF@result int@6		# jump 6, badjump 3
JUMP end							# jump 7, fwjump 4
JUMP start 						# jump 8, backjump 1
JUMP start 						# jump 9, backjump 2
JUMP start 						# jump 10, backjump 3
JUMP start 						# jump 11, backjump 4

LABEL label2						# label 4
MOVE GF@lol int@12
MOVE GF@kek int@4
IDIV GF@result GF@lol GF@kek
JUMPIFEQ label3 GF@result int@3		# jump 12, backjump 5
JUMPIFNEQ badjump4 GF@result int@3		# jump 13, badjump 4

LABEL end							# label 5
WRITE string@Program\010finished\010successfully
EXIT int@0
 