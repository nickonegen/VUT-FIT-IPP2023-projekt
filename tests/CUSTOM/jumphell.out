<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode23"><instruction order="1" opcode="LABEL"><arg1 type="label">start</arg1></instruction><instruction order="2" opcode="PUSHS"><arg1 type="int">5</arg1></instruction><instruction order="3" opcode="PUSHS"><arg1 type="int">10</arg1></instruction><instruction order="4" opcode="DEFVAR"><arg1 type="var">GF@kek</arg1></instruction><instruction order="5" opcode="DEFVAR"><arg1 type="var">GF@lol</arg1></instruction><instruction order="6" opcode="DEFVAR"><arg1 type="var">GF@result</arg1></instruction><instruction order="7" opcode="POPS"><arg1 type="var">GF@lol</arg1></instruction><instruction order="8" opcode="POPS"><arg1 type="var">GF@kek</arg1></instruction><instruction order="9" opcode="ADD"><arg1 type="var">GF@result</arg1><arg2 type="var">GF@kek</arg2><arg3 type="var">GF@lol</arg3></instruction><instruction order="10" opcode="PUSHS"><arg1 type="var">GF@result</arg1></instruction><instruction order="11" opcode="JUMPIFEQ"><arg1 type="label">label1</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">15</arg3></instruction><instruction order="12" opcode="JUMPIFNEQ"><arg1 type="label">badjump1</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">15</arg3></instruction><instruction order="13" opcode="LABEL"><arg1 type="label">label1</arg1></instruction><instruction order="14" opcode="POPS"><arg1 type="var">GF@kek</arg1></instruction><instruction order="15" opcode="MOVE"><arg1 type="var">GF@lol</arg1><arg2 type="int">3</arg2></instruction><instruction order="16" opcode="SUB"><arg1 type="var">GF@result</arg1><arg2 type="var">GF@kek</arg2><arg3 type="var">GF@lol</arg3></instruction><instruction order="17" opcode="JUMPIFEQ"><arg1 type="label">label2</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">12</arg3></instruction><instruction order="18" opcode="JUMPIFNEQ"><arg1 type="label">badjump2</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">12</arg3></instruction><instruction order="19" opcode="LABEL"><arg1 type="label">label3</arg1></instruction><instruction order="20" opcode="MOVE"><arg1 type="var">GF@kek</arg1><arg2 type="int">2</arg2></instruction><instruction order="21" opcode="MOVE"><arg1 type="var">GF@lol</arg1><arg2 type="int">3</arg2></instruction><instruction order="22" opcode="MUL"><arg1 type="var">GF@result</arg1><arg2 type="var">GF@kek</arg2><arg3 type="var">GF@lol</arg3></instruction><instruction order="23" opcode="JUMPIFEQ"><arg1 type="label">end</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">6</arg3></instruction><instruction order="24" opcode="JUMPIFNEQ"><arg1 type="label">badjump3</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">6</arg3></instruction><instruction order="25" opcode="JUMP"><arg1 type="label">end</arg1></instruction><instruction order="26" opcode="JUMP"><arg1 type="label">start</arg1></instruction><instruction order="27" opcode="JUMP"><arg1 type="label">start</arg1></instruction><instruction order="28" opcode="JUMP"><arg1 type="label">start</arg1></instruction><instruction order="29" opcode="JUMP"><arg1 type="label">start</arg1></instruction><instruction order="30" opcode="LABEL"><arg1 type="label">label2</arg1></instruction><instruction order="31" opcode="MOVE"><arg1 type="var">GF@lol</arg1><arg2 type="int">12</arg2></instruction><instruction order="32" opcode="MOVE"><arg1 type="var">GF@kek</arg1><arg2 type="int">4</arg2></instruction><instruction order="33" opcode="IDIV"><arg1 type="var">GF@result</arg1><arg2 type="var">GF@lol</arg2><arg3 type="var">GF@kek</arg3></instruction><instruction order="34" opcode="JUMPIFEQ"><arg1 type="label">label3</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">3</arg3></instruction><instruction order="35" opcode="JUMPIFNEQ"><arg1 type="label">badjump4</arg1><arg2 type="var">GF@result</arg2><arg3 type="int">3</arg3></instruction><instruction order="36" opcode="LABEL"><arg1 type="label">end</arg1></instruction><instruction order="37" opcode="WRITE"><arg1 type="string">Program\010finished\010successfully</arg1></instruction><instruction order="38" opcode="EXIT"><arg1 type="int">0</arg1></instruction></program>
